x/cosmwasmpool/pool_module.go

```diff
@@ -100,12 +100,15 @@ func (k Keeper) GetPool(ctx sdk.Context, poolId uint64) (poolmanagertypes.PoolI,
 func (k Keeper) GetPools(ctx sdk.Context) ([]poolmanagertypes.PoolI, error) {
 	return osmoutils.GatherValuesFromStorePrefix(
 		ctx.KVStore(k.storeKey), types.PoolsKey, func(value []byte) (poolmanagertypes.PoolI, error) {
-			pool := model.Pool{}
+			pool := model.CosmWasmPool{}
 			err := k.cdc.Unmarshal(value, &pool)
 			if err != nil {
 				return nil, err
 			}
-			return &pool, nil
+			return &model.Pool{
+				CosmWasmPool: pool,
+				WasmKeeper:   k.wasmKeeper,
+			}, nil
```

Comment:
Could you please check whether this breaks the poolmanager `all-pools` query?

I vaguely remember that serialization issues as being the reason for not setting the wasm keeper here at the start

Reply:
Confirming that this did not break all-pools. Here is an except from all-pools showing the CW pools:

```
- '@type': /osmosis.cosmwasmpool.v1beta1.CosmWasmPool
  code_id: "148"
  contract_address: osmo1gyg0pys40ex2f6a4dytd3ewpx2xfrsnt3rdc2t4j3s3jc9qx8kqsny066c
  instantiate_msg: eyJwb29sX2Fzc2V0X2Rlbm9tcyI6IFsiaWJjLzgyNDJBRDI0MDA4MDMyRTQ1N0QyRTEyRDQ2NTg4RkQzOUZCNTRGQjI5NjgwQzZDNzY2M0QyOTZCMzgzQzM3QzQiLCJpYmMvNEFCQkVGNEM4OTI2REREQjMyMEFFNTE4OENGRDYzMjY3QUJCQ0VGQzA1ODNFNEFFMDVENkU1QUEyNDAxRERBQiJdLCJhZG1pbiI6I>
  pool_id: "1211"
- '@type': /osmosis.cosmwasmpool.v1beta1.CosmWasmPool
  code_id: "148"
  contract_address: osmo10c8y69yylnlwrhu32ralf08ekladhfknfqrjsy9yqc9ml8mlxpqq2sttzk
  instantiate_msg: eyJwb29sX2Fzc2V0X2Rlbm9tcyI6IFsiaWJjL0QxODkzMzVDNkU0QTY4QjUxM0MxMEFCMjI3QkYxQzFEMzhDNzQ2NzY2Mjc4QkEzRUVCNEZCMTQxMjRGMUQ4NTgiLCJpYmMvNDk4QTA3NTFDNzk4QTBEOUEzODlBQTM2OTExMjNEQURBNTdEQUE0RkUxNjVENUM3NTg5NDUwNUI4NzZCQTZFNCJdLCJhZG1pbiI6I>
  pool_id: "1212"
```